### DISCUSSION TITLE
liquid evaporation model with thermocouples inside the liquid in pool fire

### DISCUSSION THREAD

**Discussion by PurnaPushkalaSomasundaram** [firemodels/fds#12868] [Q&A]
Hello, this is Purna from Politecnico di Milano, Milan. I have designed the pool fire of corn oil using FDS using the cylindrical surface of the tank.  I would like to know the temperature of the oil inside the tank by placing thermocouples. Is it possible by using FDS?

**Comment by mcgratta** [comment]
INSIDE WALL TEMPERATURE

**Reply by PurnaPushkalaSomasundaram** [reply]
So, should I include any liquid evaporation models by layering inside the tank by using MATL?

**Reply by mcgratta** [reply]
If you want to PREDICT the burning rate of the fuel, you must use the liquid evaporation model.

**Comment by PurnaPushkalaSomasundaram** [comment]
Yes,  I am predicting the burning rate of the fuel. Should I try to design each layer inside the tank using MATL and then place a device to measure temperature at each 0.01 m height of the entire height of 0.15 m tank? Does that make sense?

**Comment by mcgratta** [comment]
Your top layer should be liquid, and you can have solid layers below that. This should all be done on the SURF line. Your devices can be placed at any DEPTH.

**Comment by PurnaPushkalaSomasundaram** [comment]
okay and thank you.

**Comment by PurnaPushkalaSomasundaram** [comment]
what to do if your model's temperature is over-predicted than it should?

**Reply by drjfloyd** [reply]
Have you done a grid study?
Have you verified you have the correct inputs?
Have you examined the sensitivity of model results to the uncertainty of your input data?
Do you have sufficient outputs to evaluate what is happening in the simulation to assess the root cause of the overprediction?

**Comment by PurnaPushkalaSomasundaram** [comment]
I did the grid analysis for the pool fire study of corn oil with required MLR with the heat of combustion. I get the results when the obstruction is square and single obstruction. But when I make multiple rectangular obstructions to form the required area, the temperature shoots up.
[2af1.txt](https://github.com/firemodels/fds/files/15325024/2af1.txt)

**Reply by drjfloyd** [reply]
What is the thermal boundary condition specified for your Fire surface?  That is what inputs have you given here that allow FDS to solve for the heat transfer inside your pool?

&SURF ID='Fire ',
      COLOR='RED',
      MLRPUA=0.039028,
      RAMP_Q='Fire _RAMP_Q',
      HEAT_TRANSFER_MODEL='LOGLAW'/


This is not how to do INSIDE WALL TEMPERATURE.  The XYZ should be that of the exposed wall cell and not deep inside the OBST.

&DEVC ID='SOLID', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.0, IOR=3/
&DEVC ID='SOLID01', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.01, IOR=3/

**Comment by PurnaPushkalaSomasundaram** [comment]
Okay sir I will do it. But if I need to design the inside of the obstruction with depths , what should I do?

**Comment by PurnaPushkalaSomasundaram** [comment]
At different heights inside the obstruction, I want to know the temperature using thermococuples.

**Reply by drjfloyd** [reply]
Your solid is 
&OBST ID='1', XB=-0.35,0.35,-0.35,0.35,0.0,0.25, SURF_IDS='Fire ','INERT','INERT'/ 

upwards facing wall cells only exist at the z=0.25 m plane.  The INSIDE WALL TEMPERATURE gives the temperature at the depth as measured from the wall cell. For temperatures going down into the tank you need devices like:

&DEVC ID='1 cm depth', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.02, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/

You also need to have a temperature profile inside the solid which means providing inputs for FDS to compute heat transfer.

**Comment by PurnaPushkalaSomasundaram** [comment]
yes sure and thanks. I will model them and will show them to know if it's right. Thanks a lot sir.

**Comment by PurnaPushkalaSomasundaram** [comment]
I have a doubt. Do I have to use a layered surface inside the tank to specify the temperature profiles?

**Comment by mcgratta** [comment]
If you want to model the interior of the tank, beneath the surface of the oil, FDS may not be the best tool. It does not account for the movement of the liquid, and the heat transfer is just one-dimensional. At best, you can estimate the liquid temperature just below the surface.

**Comment by PurnaPushkalaSomasundaram** [comment]
which is the best software for modeling the interior of the tank?

**Comment by PurnaPushkalaSomasundaram** [comment]
If so I want to model with FDS, how would I implement it for surfaces beneath the oil surface using 1D heat transfer?

**Comment by mcgratta** [comment]
Your SURF line would have multiple layers. The first would be the liquid, then the solid material under the liquid.

**Comment by PurnaPushkalaSomasundaram** [comment]
&SURF ID='BaseLayer', CONDUCTIVITY=1.0, SPECIFIC_HEAT=1000.0, TMP_FRONT=20.0 /
&SURF ID='SecondLayer', CONDUCTIVITY=0.5, SPECIFIC_HEAT=800.0, TMP_FRONT=20.0 /
&SURF ID='ThirdLayer', CONDUCTIVITY=0.2, SPECIFIC_HEAT=600.0, TMP_FRONT=20.0 /
Is it like this?

**Reply by drjfloyd** [reply]
There are multiple examples of multi-layer surfaces in the User's Guide. Please review them.

**Comment by PurnaPushkalaSomasundaram** [comment]
What solid material should I use under the first layer?

**Comment by PurnaPushkalaSomasundaram** [comment]
[pool_fire1_AF.txt](https://github.com/firemodels/fds/files/15339640/pool_fire1_AF.txt)
Can you look through this file?

**Comment by PurnaPushkalaSomasundaram** [comment]
I would like to know why the temperature of this case is high and how to reduce it.
[pool_fire1_AF (1).txt](https://github.com/firemodels/fds/files/15392441/pool_fire1_AF.1.txt)

**Reply by drjfloyd** [reply]
What temperature is high?
How do you know it is high?

**Reply by drjfloyd** [reply]
Did this case actually run for you? Is this the input file as written by Pyrosim?

C:\Projects\FDS_test>fds pool_fire1_AF.1.fds

 Starting FDS ...

 MPI Process      0 started on L-PW078R66

 Reading FDS input file ...


ERROR(101): Problem with SURF number 2, line number 54 (CHID: pool_fire1_AF)

ERROR: FDS was improperly set-up - FDS stopped (CHID: pool_fire1_AF)

**Comment by PurnaPushkalaSomasundaram** [comment]
I ran the coding and found it to be high. I ran using pyrosim interface and found that the depth at 13 cm should be a heat conducting surface. What does it exactly mean?

**Reply by drjfloyd** [reply]
Is this the input file as written by Pyrosim in the directrory where you ran it?  The inputs you gave are not valid inputs for any version of FDS that I can recall going back to FDS 1.  You need to fix your inputs so it runs with the current release and post here. 

Also:

What temperature is high?
How do you know it is high?

**Comment by PurnaPushkalaSomasundaram** [comment]
I am attaching the file from which I began to develop the layers and inside temperature. The flame temperature is fine at the moment using the attached file below.
[corn pool_fire_AF.txt](https://github.com/firemodels/fds/files/15438524/corn.pool_fire_AF.txt)
 I have problem in constructing the layers of liquid on the top and then solid layers. I want to plan for liquid on the surface using ramp temperature and the last layer with water layer of 0.01m and then corresponding top layers with corn oil.

**Reply by mcgratta** [reply]
It appears that you are just specifying the mass loss rate of the liquid. There is no computation of what is below this surface. You have just created a gas burner. I do not understand what you are trying to do here.

**Comment by PurnaPushkalaSomasundaram** [comment]
I aim to model a solid obstruction (tank) with layers below it to find the inside wall temperature temperature within the solid.

**Reply by mcgratta** [reply]
I do not understand what you mean. Draw a very simple sketch of what you hope to do.

**Comment by PurnaPushkalaSomasundaram** [comment]
<img width="218" alt="image" src="https://github.com/firemodels/fds/assets/166021559/d68ae671-fe2b-4ce5-bc0e-dc606d583214">
this is my solid obstruction of a 0.8 m diameter tank with 0.25 cm height corn oil pool fire tank with the surface as a burner with corn oil properties. I want to know the temperature inside the tank for every 0.01 m height using the inside wall temperature using this syntax.

&DEVC ID='1 cm depth', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.02, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/

I know that the top layer is liquid and layers of solid needs to be within the tank. Can I use the same property of corn oil in solids for each layer?

**Comment by mcgratta** [comment]
```
&SURF ID='Fire',
      COLOR='RED',
      MLRPUA=0.039028,
      RAMP_Q='Fire_RAMP_Q',
      TMP_FRONT=140.0/
```
This `SURF` line indicates that you are specifying the burning rate of the liquid and that you are specifying the surface temperature. FDS will not do an in-depth heat conduction calculation because there is no need for it. You are already telling FDS what to do. If you want an in-depth temperature, you must use the burning liquid pyrolysis model.

**Comment by PurnaPushkalaSomasundaram** [comment]
I tried to model in this way the liquid pyrolysis model of corn oil to have  in-depth temperature. Is it correct?

&HEAD CHID='corn_oil_pyrolysis', TITLE='Corn Oil Pyrolysis Simulation' /

&MESH IJK=100,80,80, XB=-2.5,2.5,-2.5,2.5,0.0,4.0  /

&TIME T_END=5400.0 /

&REAC ID='Corn Oil Pyrolysis',
      FUEL='Corn Oil',
      C=22.0, H=25, O=6,N=1
      NU_O2=1.0,
      CO_YIELD=0.025,
      SOOT_YIELD=0.08,
      HCN_YIELD=0.01,
      HEAT_OF_COMBUSTION=3.7E+4//

&MATL ID='Corn Oil',
      FYI='Liquid fuel for pyrolysis',
      SPECIFIC_HEAT=2.039,
      CONDUCTIVITY=0.152,
      DENSITY=920,
      HEAT_OF_REACTION=300,
      PYROLYSIS_RATE=10,
      IGNITION_TEMPERATURE=623.15,
      BOILING_TEMPERATURE=573.15 /

&SURF ID='Corn Oil Surface',
      COLOR='YELLOW',
      HRRPUA=1585.71,
      MATERIAL='Corn Oil',
      TMP_FRONT=140/

&OBST XB=-0.35,0.35,-0.35,0.35,0.0,0.25, SURF_IDS='Corn Oil Surface' /


&INIT ID='Initial Conditions',
      TEMPERATURE=140.0 /

&VENT ID='Mesh Vent: Mesh01 [XMAX]', SURF_ID='OPEN', XB=2.5,2.5,-2.5,2.5,0.0,4.0/ 
&VENT ID='Mesh Vent: Mesh01 [XMIN]', SURF_ID='OPEN', XB=-2.5,-2.5,-2.5,2.5,0.0,4.0/ 
&VENT ID='Mesh Vent: Mesh01 [YMAX]', SURF_ID='OPEN', XB=-2.5,2.5,2.5,2.5,0.0,4.0/ 
&VENT ID='Mesh Vent: Mesh01 [YMIN]', SURF_ID='OPEN', XB=-2.5,2.5,-2.5,-2.5,0.0,4.0/ 
&VENT ID='Mesh Vent: Mesh01 [ZMAX]', SURF_ID='OPEN', XB=-2.5,2.5,-2.5,2.5,4.0,4.0/

&DEVC ID='1 cm depth', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.02, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='1 cm depth', DEPTH=0.03, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.04, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='1 cm depth', DEPTH=0.05, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.06, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/&DEVC ID='1 cm depth', DEPTH=0.01, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.07, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='1 cm depth', DEPTH=0.08, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.09, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='1 cm depth', DEPTH=0.1, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.11, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/ 
&DEVC ID='1 cm depth', DEPTH=0.12, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.13, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='1 cm depth', DEPTH=0.14, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/
&DEVC ID='2 cm depth', DEPTH=0.15, QUANTITY='INSIDE WALL TEMPERATURE', XYZ=0.0,0.0,0.25, IOR=3/

&TAIL /

**Comment by mcgratta** [comment]
```
PYROLYSIS_RATE=10,
IGNITION_TEMPERATURE=623.15,
BOILING_TEMPERATURE=573.15 /
```
You should only set `BOILING_TEMPERATURE`

**Reply by PurnaPushkalaSomasundaram** [reply]
&MATL ID='Corn Oil',
      FYI='Liquid fuel for pyrolysis',
      SPECIFIC_HEAT=2.039,
      CONDUCTIVITY=0.152,
      DENSITY=920,
      HEAT_OF_REACTION=300,
      BOILING_TEMPERATURE=573.15 /
Is this right?

**Comment by drjfloyd** [comment]
&SURF ID='Corn Oil Surface',
COLOR='YELLOW',
HRRPUA=1585.71,
MATERIAL='Corn Oil',
TMP_FRONT=140/

You are still prescribing the HRRPUA and TMP_FRONT. 
Does anyplace in the User's Guide state that  MATERIAL is a valid input for SURF?

**Comment by PurnaPushkalaSomasundaram** [comment]
It does not say in any place that material is keyword. So, on the SURF ID should I mention only the external flux value?
